exist-core/src/main/java/org/exist/xquery/value/FunctionParameterFunctionSequenceType.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.xquery.value;
+
+import org.exist.xquery.Cardinality;
+import org.exist.xquery.ErrorCodes;
+import org.exist.xquery.XPathException;
+import org.exist.xquery.FunctionSignature;
+
+/**
+ * This class is used to specify the name and description of an XQuery function parameter of type function.
+ */
+public class FunctionParameterFunctionSequenceType extends FunctionParameterSequenceType {
+
+    private SequenceType[] parameters = null;
+    private SequenceType returnType = new SequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE);
+
+    /**
+     * mimics isOverloaded property of FunctionSignature
+     * @see org.exist.xquery.FunctionSignature
+     */
+    private int arity = -1;
```

Comment:
It is not clear to me how this relates to `isOverloaded` could you add some information? It is not possible to have `overloaded` function signatures in XQuery, this was previously just a non-compliant thing offered by eXist-db that was later removed.

Could you add some explanation of why this is needed? It looks to me that the arity of the function (as a a parameter) is simply the value of `parameterTypes.length`, and that we don't need an additional class member variable for that.